openff/toolkit/topology/molecule.py

```diff
@@ -596,8 +598,7 @@ def _position(self, atom_positions):
 
         originwt, xdir, ydir = self.virtual_site.local_frame_weights
         disp = self.virtual_site.local_frame_position
-        _unit = disp.unit
-        x, y, z = disp / _unit
+        x, y, z = disp.value_in_unit(disp.unit)
```

Comment:
I'm not sure how you feel about using private attributes, but this could be written more concisely as `x, y, z = disp._value` (analogous to `pint.Quantity.magnitude`).

Alternatively, you could try `x, y, z = disp.value_in_unit(unit.angstrom)` or `x, y, z = disp.value_in_unit(atom_positions.unit)` and see if anything breaks. If I understand correctly -- @trevorgokey please correct me -- this whole chain of `local_frame_position` etc. seems to rely on the unit implicitly being `unit.angstrom`, since all the documentation refers to angstrom. For the most part it doesn't really affect the output of the function, except for this one, where the units of the input `atom_positions` really should match the units of `self.virtual_site.local_frame_position` for the output to make sense. But that might be outside the scope of this PR.

Either way LGTM!

Reply:
Yes,  `atom_positions` is actually positions (L668), so the docstring is outdated. It appears that, if we don't assume anything about any units, that yes we should use `disp.value_in_unit(atom_positions.unit)` since it will later be multiplied by `*axis` (L626), which takes its units from the atoms (and `origin` which also takes its units from the atoms).

Overall, I do suppose `x, y, z = disp.value_in_unit(disp.unit)` is a bug if the unit is different from the atoms. Nice catch!

Reply:
```suggestion
        x, y, z = disp.value_in_unit(atom_positions.unit)
```